src/Kiota.Builder/KiotaBuilder.cs

```diff
@@ -366,12 +366,13 @@ private static void CreateMethod(string propIdentifier, string propType, CodeCla
     }
     private static void AddPathParametersToMethod(OpenApiUrlTreeNode currentNode, CodeMethod methodToAdd, bool asOptional) {
         foreach(var parameter in currentNode.GetPathParametersForCurrentSegment()) {
+            var codeName = parameter.Name.SanitizeParameterNameForCodeSymbols();
             var mParameter = new CodeParameter {
-                Name = parameter.Name,
+                Name = codeName,
                 Optional = asOptional,
                 Description = parameter.Description,
                 Kind = CodeParameterKind.Path,
-                UrlTemplateParameterName = parameter.Name,
+                SerializationName = parameter.Name.Equals(codeName) ? default : parameter.Name.SanitizeParameterNameForUrlTemplate(),
```

Comment:
I'm trying to reduce the amount of code we're generating. The code to check whether the attribute/annotation is present it limited to one place per language.

Reply:
What would the difference between always setting it so that SerializationName is never null vs only setting it when it doesn't match the `codeName`? I assumed since `CodeParameter` is a CodeDOM type, setting the property here won't directly affect the generated code. My thought was that setting it always means we can remove a bunch of `string.IsNullOr...` checks. Did I misunderstand the intent of this code?

Reply:
string.IsNullOrEmpty is o(1), string.Equals is at best o(log(n))

Reply:
In this strategy, you need to call both `string.Equals` when setting the value and `string.IsNullOrEmpty`. If it's always set, you won't need either. My worry is the number of checks that anyone reading from this property needs to make. If we remove that need for null check, we reduce the likelihood of bugs, don't you think?

Reply:
Ah, I think I'd missed the connection to a generated annotation. So if it doesn't have a value, you don't generate an annotation? I think it makes more sense now why you went with this. A boolean property would be a good idea I think.

Reply:
yeah the idea being:
- if the parameter name is a valid "code symbol" we generate the property as-is without an annotation
- if the parameter name contains exotic characters, we clean that up for the code symbol and use the serialization name for the annotation.

Reply:
done